tests/php/test-class-customize-snapshot-manager.php

```diff
@@ -729,4 +729,40 @@ public function test_replace_customize_link() {
 		$parsed_preview_url = wp_parse_url( $query_params['url'] );
 		$this->assertArrayNotHasKey( 'query', $parsed_preview_url );
 	}
+
+	/**
+	 * Test filter_user_has_cap.
+	 *
+	 * @covers \CustomizeSnapshots\Customize_Snapshot_Manager::filter_user_has_cap()
+	 */
+	function test_filter_user_has_cap() {
+		$this->mark_incompatible();
+		$post_id = $this->factory()->post->create( array(
+			'post_type' => 'post',
+			'post_status' => 'auto-draft',
+		) );
```

Comment:
Can there be other posts here including ones that have the `draft`, `private`, and `publish` statuses? Then below in the changeset data these posts can have the `post_status` overridden to for example reverse the statuses. Then at the end it can set the current user to anonymous, to an author, and to an administrator and for each then the `read_post` cap check can be verified to return the proper result for the customized state?